app/src/main/java/com/andrew/tasky/agenda/data/agenda/AgendaRepositoryImpl.kt

```diff
@@ -2,32 +2,93 @@ package com.andrew.tasky.agenda.data.agenda
 
 import android.util.Log
 import com.andrew.tasky.agenda.data.database.AgendaDatabase
-import com.andrew.tasky.agenda.data.reminder.ReminderApi
 import com.andrew.tasky.agenda.data.util.*
 import com.andrew.tasky.agenda.domain.AgendaRepository
 import com.andrew.tasky.agenda.domain.ReminderRepository
 import com.andrew.tasky.agenda.domain.models.AgendaItem
 import com.andrew.tasky.auth.data.AuthResult
 import com.andrew.tasky.auth.util.getAuthResult
 import java.time.LocalDate
+import java.time.LocalDateTime
 import java.time.LocalTime
 import java.util.*
+import kotlinx.coroutines.*
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.map
 
 class AgendaRepositoryImpl(
     private val agendaApi: AgendaApi,
     private val reminderRepository: ReminderRepository,
     private val db: AgendaDatabase,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
 ) : AgendaRepository {
 
-    override fun getAgendaItems(): Flow<List<AgendaItem>> {
-        val localReminders = db.getReminderDao().getRemindersAsFlow().map {
+    override fun getAgendaItems(localDate: LocalDate): Flow<List<AgendaItem>> {
+
+        val calendar = Calendar.getInstance()
+        calendar.set(
+            localDate.year,
+            localDate.monthValue - 1,
+            localDate.dayOfMonth
+        )
+        calendar.set(Calendar.HOUR_OF_DAY, 0)
+        calendar.set(Calendar.MINUTE, 0)
+        calendar.set(Calendar.SECOND, 0)
+        calendar.set(Calendar.MILLISECOND, 0)
+        val startEpochMilli = calendar.timeInMillis
+
+        calendar.add(Calendar.DATE, 1)
+        val endEpochMilli = calendar.timeInMillis
+
+        CoroutineScope(ioDispatcher).launch {
```

Comment:
I'd avoid launching coroutines outside of ViewModels. These are uncontrollable by the ViewModel. Imagine, your VM calls this function and the user navigates away right after. Then this call shouldn't resume, which it would with a custom scope here though.
If the user navigates back `viewModelScope` is cleared and therefore all suspending calls in it would be cancelled. Long story short, only expose suspend functions unless in a ViewModel or you're creating an object that has an actual lifecycle different than the ViewModel (which is not the case for repos)